testcases/kernel/syscalls/fsetxattr/fsetxattr02.c

```diff
@@ -0,0 +1,250 @@
+#include <sys/socket.h>
+#include <sys/un.h>
+#ifdef HAVE_SYS_XATTR_H
+# include <sys/xattr.h>
+#endif
+#include "tst_test.h"
+
+#ifdef HAVE_SYS_XATTR_H
+#define XATTR_TEST_KEY "user.testkey"
+#define XATTR_TEST_VALUE "this is a test value"
+#define XATTR_TEST_VALUE_SIZE 20
+
+#define OFFSET    11
+#define FILENAME "fsetxattr02testfile"
+#define DIRNAME  "fsetxattr02testdir"
+#define SYMLINK  "fsetxattr02symlink"
+#define FIFO     "fsetxattr02fifo"
+#define CHR      "fsetxattr02chr"
+#define BLK      "fsetxattr02blk"
+#define SOCK     "fsetxattr02sock"
+
+struct test_case {
+	char *fname;
+	int fd;
+	int fflags;
+	char *key;
+	char *value;
+	size_t size;
+	int flags;
+	int exp_err;
+	int issocket;
+};
+static struct test_case tc[] = {
+	{			/* case 00, set attr to reg */
+	 .fname = FILENAME,
+	 .fflags = O_RDONLY,
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = 0,
+	 },
+	{			/* case 01, set attr to dir */
+	 .fname = DIRNAME,
+	 .fflags = O_RDONLY,
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = 0,
+	 },
+	{			/* case 02, set attr to symlink */
+	 .fname = SYMLINK,
+	 .fflags = O_RDONLY,
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = EEXIST,
+	 },
+	{			/* case 03, set attr to fifo */
+	 .fname = FIFO,
+	 .fflags = (O_RDONLY | O_NONBLOCK),
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = EPERM,
+	 },
+	{			/* case 04, set attr to character special */
+	 .fname = CHR,
+	 .fflags = O_RDONLY,
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = EPERM,
+	 },
+	{			/* case 05, set attr to block special */
+	 .fname = BLK,
+	 .fflags = O_RDONLY,
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = EPERM,
+	 },
+	{			/* case 06, set attr to socket */
+	 .fname = SOCK,
+	 .fflags = O_RDONLY,
+	 .key = XATTR_TEST_KEY,
+	 .value = XATTR_TEST_VALUE,
+	 .size = XATTR_TEST_VALUE_SIZE,
+	 .flags = XATTR_CREATE,
+	 .exp_err = EPERM,
+	 .issocket = 1,
+	 },
+};
+
+static void verify_fsetxattr(unsigned int i)
+{
+	TEST(fsetxattr(tc[i].fd, tc[i].key, tc[i].value, tc[i].size,
+			tc[i].flags));
+
+	if (TEST_RETURN == -1 && TEST_ERRNO == EOPNOTSUPP)
+		tst_brk(TCONF, "fsetxattr() not supported");
+
+	if (!tc[i].exp_err) {
+		if (TEST_RETURN) {
+			tst_res(TFAIL | TTERRNO,
+				"fsetxattr() on %s failed with %li",
+				tc[i].fname + OFFSET, TEST_RETURN);
+			return;
+		}
+
+		tst_res(TPASS, "fsetxattr() on %s passed",
+				tc[i].fname + OFFSET);
+		return;
+	}
+
+	if (TEST_RETURN == 0) {
+		tst_res(TFAIL, "fsetxattr() on %s passed unexpectedly",
+				tc[i].fname + OFFSET);
```

Comment:
We can probably shorten the filenames, since the test temporary directory is unique and the name is based on the test name. That way we don't have to use the offset, but either way it's fine I guess.

Reply:
yep, thought it was long also. got it from setxattr()/getxattr() tests. will keep like that for the sake of being equal to other tests only since its fine.